### ISSUE TITLE
Make dark mode optional (disabled by default?)

### ISSUE THREAD

**Issue by purplespider** [silverstripe/silverstripe-login-forms#51]
While I do appreciate the extra effort that has been made to add dark mode to this module. I find it quite a jarring experience when neither the website or CMS support dark mode, as you end up with a login screen which looks completely different to both.

I'd like to be able to easily disable/enable the dark mode feature so that if the website supports it (or already uses dark styles) I can enable it, and for sites that don't, leave it disabled for a consistent experience from site to CMS.

Once the CMS support dark mode, then it would make sense for it to be enabled by default (by that I mean if the user has dark mode enabled, then display the dark theme).

**Comment by robbieaverill** [comment]
You may want to use `<meta name="color-scheme" content="only light">` to disable it (in your project code somewhere). See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta.

_Edit:_ if there's a bug preventing you from using this in your project, then let's reopen this as a bug.

**Comment by Cheddam** [comment]
@robbieaverill The `color-scheme` meta tag doesn't have any impact on whether the `prefers-color-scheme` media query is respected by the browser, which means short of adding extra CSS that cancels out the variable overrides in that media query, there's no way to disable dark mode at the moment.

I'm not sure about disabling dark mode _by default_, but we should definitely make it possible to turn off. We could do this by shifting the variable overrides inside a class on `body` (e.g. `.enable-dark-mode`), and making the presence of that class conditional based on config.

**Comment by purplespider** [comment]
> @robbieaverill The `color-scheme` meta tag doesn't have any impact on whether the `prefers-color-scheme` media query is respected by the browser

I discovered this when I tried it out!

> I'm not sure about disabling dark mode _by default_, but we should definitely make it possible to turn off. We could do this by shifting the variable overrides inside a class on `body` (e.g. `.enable-dark-mode`), and making the presence of that class conditional based on config.

That sounds perfect!

**Comment by michalkleiner** [comment]
Just a question on the side — the image displaying in the README is _not_ what you get by default when you install this module?

**Comment by purplespider** [comment]
> the image displaying in the README is _not_ what you get by default when you install this module?

It is. Unless you are viewing the page in a browser which supports "dark mode" and has it enabled, then you get the dark version of it.

My argument is that "dark mode" shouldn't display unless the above is true _and_ either the website or CMS _also_ supports a dark mode theme.

**Comment by elliot-sawyer** [comment]
Same problem here, if the user has Dark mode configured through the OS, they end up with a login screen that's entirely unlike the rest of your site (assuming you haven't implemented a dark mode theme previously). If your users are security conscious, it would appear to be a MITM attack (especially since the URL is your `/Security/login` link)

@purplespider if you need to disable it, I ended up doing this. The variables are copy-pasted from the Firefox inspector prior to dark mode being applied:
```php
            if($this->URLSegment == 'Security') {
                Requirements::customCSS("
                @media (prefers-color-scheme: dark) {
                    :root {
                        --color-button-secondary-background: transparent;
                        --color-button-secondary-hover-background: #e7ebf0;
                        --color-button-secondary-text: #0071c4;
                        --color-button-secondary-hover-text: #0071c4;
                        --color-button-primary-background: #008a00;
                        --color-button-primary-hover-background: #006400;
                        --color-button-primary-text: #fff;
                        --color-button-primary-hover-text: #fff;
                        --color-field-background: #fff;
                        --color-field-border: #ced5e1;
                        --color-field-text: #303b4d;
                        --color-form-background: #fff;
                        --color-form-text: #43536d;
                        --color-link-primary: #0071c4;
                        --color-message-error: #71141f;
                        --color-message-warning: #856404;
                        --color-page-background: #f7f8fa;
                        --color-page-text: #303b4d;
                    }
                }");
            }
```

@Cheddam If you move all the `prefers-color-scheme` rules into their own stylesheet and add it with a `<% require ... %>` block, devs can call `Requirements::block(..)` in PageController.php to block it from being loaded. That still gives you dark mode included by default, and devs have an easier option to disable it

**Comment by ScopeyNZ** [comment]
Happy to accept PRs for this enhancement 🙂

**Comment by elliot-sawyer** [comment]
@purplespider I have created a fix for this if you want to try it out: https://github.com/silverstripe/silverstripe-login-forms/pull/55

**Comment by purplespider** [comment]
Amazing! Thanks @elliot-sawyer, I did take a look at trying this myself but had trouble getting the tooling working.

Looks good. Only issue is that the white version of the Silverstripe logo still seems to be displayed, and so is barely legible:

<img width="455" alt="Screenshot 2020-02-18 at 10 41 29" src="https://user-images.githubusercontent.com/329880/74728829-5cb2ef00-523b-11ea-95a9-1e59ef31ce20.png">

Not sure if there is an easy solution to this?

**Comment by elliot-sawyer** [comment]
SilverStripeLogo.ss is actually an SVG with `@media (prefers-color-scheme: dark) { ... }` embedded in it. Not sure what the best solution for that is. 

The first thing I usually do with this module is remove the logo by creating `app/templates/SilverStripeLogo.ss` and leaving it empty.

**Comment by elliot-sawyer** [comment]
@purplespider I found a workaround, could you have another look?

**Comment by purplespider** [comment]
Nice one! Looks perfect to me!

**Comment by Cheddam** [comment]
Fixed in #55 by @elliot-sawyer - thanks so much for contributing!
